src/game_api/entity.hpp

```diff
@@ -211,6 +211,31 @@ class Entity
         return topmost;
     }
 
+    bool overlaps_with(float rect_left, float rect_bottom, float rect_right, float rect_top)
+    {
+        const auto [posx, posy] = position();
+        const float left = posx - hitboxx + offsetx;
+        const float right = posx + hitboxy + offsetx;
+        const float bottom = posy - hitboxy + offsety;
+        const float top = posy + hitboxy + offsety;
+
+        return left < rect_right
+            && rect_left < right
+            && bottom < rect_top
+            && rect_bottom < top;
+    }
+
+    bool overlaps_with(Entity* other)
+    {
+        const auto [other_posx, other_posy] = other->position();
+        const float other_left = other_posx - other->hitboxx + other->offsetx;
+        const float other_right = other_posx + other->hitboxy + other->offsetx;

```

Comment:
Same here? hitboxx instead of y

Reply:
All my tests worked because many entities have similar x and y  😅